query/sources/common/src/main/java/org/codice/ddf/admin/sources/utils/SourceUtilCommons.java

```diff
@@ -140,27 +161,50 @@ public static SourceInfoField createSourceInfoField(boolean isAvailable,
                         .values()
                         .stream())
                 .map(mapper)
-                .forEach(config -> sourceInfoListField.add(createSourceInfoField(false, config)));
+                .forEach(sourceInfoListField::add);
 
-        populateSourceAvailability(sourceInfoListField.getList(), serviceReader);
         return sourceInfoListField;
     }
 
-    private static void populateSourceAvailability(List<SourceInfoField> sourceInfoList,
-            ServiceReader serviceReader) {
-        List<Source> sources = getAllSourceReferences(serviceReader);
-        for (SourceInfoField sourceInfoField : sourceInfoList) {
-            for (Source source : sources) {
-                if (source instanceof ConfiguredService) {
-                    ConfiguredService service = (ConfiguredService) source;
-                    if (service.getConfigurationPid()
-                            .equals(sourceInfoField.config()
-                                    .pid())) {
-                        sourceInfoField.isAvaliable(source.isAvailable());
-                        break;
-                    }
+    public Report saveSource(PidField pid, Map<String, Object> serviceProps, String factoryPid) {
```

Comment:
We should talk about whether it's worth breaking up the save sources functions into `create` and `update` instead